lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -341,6 +341,7 @@ protected function gatherRowData(array $data, array &$id, array &$nonemptyCompon
             foreach ($rowData['newObjects'] as $objIndex => $newObject) {
                 $class  = $newObject['class'];
                 $args   = $newObject['args'];
+                ksort($args);
```

Comment:
Thanks.

Is there any test case for this? I would like to add it to test suite.